astropy/conftest.py

```diff
@@ -28,6 +28,7 @@
 except ImportError:
     pass
 else:
+    matplotlib.rcParams.update(matplotlib.rcParamsDefault)
```

Comment:
Does this change the matplotlib settings forever or just duration of the tests? For example, if I run the tests locally, I wouldn't want the test to change my matplotlib settings for real.

Reply:
The matplotlibrc file is read when matplotlib is imported for the first time (which in tests is here, I think). This then overrides the loaded rc parameters with the defaults and will persist until the end of the python session. So if for some reason you're running interactively, run the tests, then want to plot something, it will use the default rc parameters, but if you start a new interpreter and import matplotlib, it will seek a matplotlibrc file as normal (https://matplotlib.org/users/customizing.html).

If this isn't acceptable, the other way is to set an environment variable MATPLOTLIBRC containing the directory of the matplotlibrc file to read (https://matplotlib.org/users/customizing.html). If this can be set to point to an appropriate matplotlibrc file (to be added to the repo), that would also work. I don't know how to manipulate the testing environment, though, and just resetting to defaults is better, I think, since it will track changes to the default settings from the matplotlib side, which we should probably care about.

Reply:
The use of `setup_module` and `teardown_module` in the test script (not at `conftest.py`) might limit the scope if this change. However, I'll leave this for @astrofrog to decide. Thanks!

Reply:
@astrofrog Alright, pushed a variation which caches the matplotlib.rcParams before the tests, reverts to official matplotlib defaults, then puts everything back the way it was after the tests.